core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -130,6 +132,32 @@ package object experimental {
     }
   }
 
+  /** This class provides the `Lit` method needed to define a `Vec` literal
+    */
+  object VecLiterals {
+    implicit class AddVecLiteralConstructor[T <: Vec[_]](x: T) {
+      /** Given a generator of a list tuples of the form [Int, Data]
+        * constructs a Vec literal, parallel concept to `BundleLiteral`
+        *
+        * @param elems generates a lit of `(Int, Data)` where the I
+        * @return
+        */
+      def Lit(elems: (Int, Data)*): T = {
+        x._makeLit(elems: _*)
+      }
+    }
+
+    implicit class AddObjectLiteralConstructor[T <: Vec.type](x: T) {
```

Comment:
Why is this in a different implicit class from the above?
Is it for Vec as an object? A comment here might be helpful, eg this infers based on type.

Reply:
Yes, this for handling a different style of constructor that does not require providing each index.
I have added a comment to this effect